rust/connlib/snownet/src/node.rs

```diff
@@ -164,14 +166,14 @@ where
         packet: &[u8],
         now: Instant,
         buffer: &'s mut [u8],
-    ) -> Result<Option<(TId, IpPacket<'s>)>, Error> {
+    ) -> Result<Option<(TId, MutableIpPacket<'s>)>, Error> {
         // First, check if a `StunBinding` wants the packet
```

Comment:
@conectado I am wondering, should we automatically add `local` as a host-candidate here? If we are receiving traffic on this socket, then this is most likely one of the best sockets to use.

We could then even remove `add_local_interface`, assuming we always have at least one STUN and one TURN server to talk to AND we aren't trying to connect to a gateway that is reachable over the local network but without connectivity to a STUN & TURN server.

I think that could even remove our dependency on `if-watcher`!

Reply:
I convinced myself that this is an exceptional idea and implemented it: https://github.com/firezone/firezone/pull/3528